cmd/cmdexecutor/cmdexecutor.go

```diff
@@ -225,9 +228,24 @@ func getPodNamesUsingLabelSelector(labelSelector, namespace string) ([]types.Nam
 		return nil, err
 	}
 
-	pods, err := core.Instance().GetPods(namespace, selectorsMap)
-	if err != nil {
-		return nil, err
+	var pods *v1.PodList
+
+	for i := 0; i < maxRetries; i++ {
+		pods, err = core.Instance().GetPods(namespace, selectorsMap)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(pods.Items) > 0 {
+			break
+		}
+
+		logrus.Infof("no pods found yet in namespace: %s with label selector: %s", namespace, labelSelector)
+		time.Sleep(time.Duration(retryInterval) * time.Second)
+	}
+
+	if len(pods.Items) == 0 {
```

Comment:
Let's not error out if pod list is empty.

Reply:
Done. PTAL


```
time="2024-05-14T15:49:56Z" level=info msg="Running pod command executor: 99.9.9-ce904ebc6"
time="2024-05-14T15:49:56Z" level=info msg="Using timeout: 86400 seconds"
time="2024-05-14T15:49:56Z" level=info msg="no pods found yet in namespace: pg1 with label selector: deployments.pds.portworx.com/pod-type=statefulset,role=master,pds/deployment-name=pg-samore-pg-1-vr7lln"
time="2024-05-14T15:50:11Z" level=info msg="no pods found yet in namespace: pg1 with label selector: deployments.pds.portworx.com/pod-type=statefulset,role=master,pds/deployment-name=pg-samore-pg-1-vr7lln"
time="2024-05-14T15:50:26Z" level=info msg="no pods found yet in namespace: pg1 with label selector: deployments.pds.portworx.com/pod-type=statefulset,role=master,pds/deployment-name=pg-samore-pg-1-vr7lln"
time="2024-05-14T15:50:41Z" level=info msg="no pods found yet in namespace: pg1 with label selector: deployments.pds.portworx.com/pod-type=statefulset,role=master,pds/deployment-name=pg-samore-pg-1-vr7lln"
time="2024-05-14T15:50:56Z" level=info msg="no pods found yet in namespace: pg1 with label selector: deployments.pds.portworx.com/pod-type=statefulset,role=master,pds/deployment-name=pg-samore-pg-1-vr7lln"
time="2024-05-14T15:51:11Z" level=info msg="no pods found yet in namespace: pg1 with label selector: deployments.pds.portworx.com/pod-type=statefulset,role=master,pds/deployment-name=pg-samore-pg-1-vr7lln"
time="2024-05-14T15:51:26Z" level=info msg="Checking status on command: pds-backup.sh --type=snapshot && ((sleep 10; touch /tmp/killme-cbfe5e37-d4fa-447f-89d6-06414fb96e64) &) && ${WAIT_CMD}"
time="2024-05-14T15:51:26Z" level=info msg="no pods found in namespace: pg1 with label selector: deployments.pds.portworx.com/pod-type=statefulset,role=master,pds/deployment-name=pg-samore-pg-1-vr7lln"
time="2024-05-14T15:51:26Z" level=info msg="exiting successfully without running any commands"
```